x/stablestake/keeper/msg_server_bond.go

```diff
@@ -77,11 +81,11 @@ func (k msgServer) Bond(goCtx context.Context, msg *types.MsgBond) (*types.MsgBo
 		return nil, err
 	}
 
-	params.TotalValue = params.TotalValue.Add(msg.Amount)
-	k.SetParams(ctx, params)
+	pool.TotalValue = pool.TotalValue.Add(msg.Amount)
```

Comment:
We need to update redemption_rate as well for the pool, or we can remove redemption_rate member from pool data structure and calculate it whenever its required, that way there is less chance of using incorrect values as we will be calculating every single time